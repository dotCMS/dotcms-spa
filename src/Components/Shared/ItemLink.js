import React from 'react';
import { Link } from 'react-router-dom';

const ItemLink = (props) => {
    return (
        <Link
            className={props.className}
            to={{
                pathname: props.pathname + window.location.search,
                state: props.state
            }}
        >
            {props.children}
        </Link>
    );
};

export default ItemLink;
