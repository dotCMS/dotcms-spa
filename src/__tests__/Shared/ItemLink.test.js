import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import ItemLink from '../../Components/Shared/ItemLink';

describe('<ItemLink />', () => {
    it('renders and matches snapshot', () => {
        window.history.pushState({}, 'Test Title', '/test.html?lang=en');
        const props = {
            className: 'testCss',
            pathname: 'testPath',
            state: {
                id: 'testId',
                name: 'testName'
            }
        };
        const wrapper = shallow(<ItemLink {...props} />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
});
