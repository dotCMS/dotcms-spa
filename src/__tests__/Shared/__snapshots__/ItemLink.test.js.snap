// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemLink /> renders and matches snapshot 1`] = `
<Link
  className="testCss"
  replace={false}
  to={
    Object {
      "pathname": "testPath?lang=en",
      "state": Object {
        "id": "testId",
        "name": "testName",
      },
    }
  }
/>
`;
